lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -1446,35 +1547,20 @@ ParseResult FIRStmtParser::parsePostFixFieldId(Value &result) {
            << result.getType();
   auto indexNo = *indexV;
 
-  // Make sure the field name matches up with the input value's type and
-  // compute the result type for the expression.
-  NamedAttribute attrs = {getConstants().fieldIndexIdentifier,
-                          builder.getI32IntegerAttr(indexNo)};
-  auto resultType = SubfieldOp::inferReturnType({result}, attrs, {});
-  if (!resultType) {
-    // Emit the error at the right location.  translateLocation is expensive.
-    (void)SubfieldOp::inferReturnType({result}, attrs, translateLocation(loc));
-    return failure();
-  }
-
-  // Check if we already have created this Subindex op.
-  auto &value = moduleContext.getCachedSubaccess(result, indexNo);
-  if (value) {
-    result = value;
-    return success();
+  FailureOr<Value> subResult;
+  if (isa<RefType>(result.getType())) {
+    NamedAttribute attrs = {getConstants().indexIdentifier,
+                            builder.getI32IntegerAttr(indexNo)};
+    subResult = emitCachedSubAccess<RefSubOp>(result, attrs, indexNo, loc);
```

Comment:
RefSubOp is not supported by LowerXMR (in this general use, it works for current user, memtaps), but due to workaround for memtaps we always lower ref-of-aggregate entirely in LowerTypes so this is not an issue even with preserving aggregates otherwise.